external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchStream.scala

```diff
@@ -73,6 +74,11 @@ private[kafka010] class KafkaMicroBatchStream(
 
   private var endPartitionOffsets: KafkaSourceOffset = _
 
+  private lazy val offsetMetrics = new KafkaCommitsSource()
+
+  private val shouldSetCommitOffsetsOnCheckpoints =
+    Option(options.get("setCommitOffsetsOnCheckpoints")).exists(_.toBoolean)
```

Comment:
KafkaSourceProvider looks like a place to define configuration key - could you add it instead of using string constant here?

Reply:
ok sure

Reply:
@HeartSaVioR I will remove the prefix as there is enough added from the metricSystem: https://github.com/prometheus/jmx_exporter/blob/master/example_configs/spark.yml